system/threads.cpp

```diff
@@ -39,6 +39,29 @@ void _start();
 
 std::atomic<uint32_t*> mainThreadWaitAddress = 0;
 
+uint32_t wakeThreadsFutex(uint32_t* uaddr, int amount)
+{
+	uint32_t threadsWokenUp = 0;
+	// If the main thread is waiting on this address, handle this case specially.
+	futexSpinLock.lock();
+	if (uaddr == mainThreadWaitAddress.load())
+	{
+		// Notify the main thread that it can wake up.
+		mainThreadWaitAddress.store(0);
+		amount -= 1;
+		threadsWokenUp = 1;
+		// If there are no other threads to wake up, return here.
+		if (amount <= 0)
```

Comment:
can we unnest this logic and deduplicate the spinlock unlocking?
just put a `if(amount==0)return` after the unlock outside of the if

Reply:
I see that it was already like this before refactoring into a function, but let's improve it

Reply:
I've changed it to now be outside of the if, so there's only 1 unlock